tutorials/event_sampler_tutorial.ipynb

```diff
@@ -0,0 +1,582 @@
+    "sky_model_pntgaus = SkyModel(spectral_model=spectral_model_pwl, spatial_model=spatial_model_gauss)\n",
+    "\n",
+    "models_pntgaus = Models([sky_model_pntgaus])\n",
+    "\n",
+    "file_model = \"gauss-pwl.yaml\"\n",
+    "models_pntgaus.write(file_model, overwrite=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We can read the `Dataset` already created and fill it with the new `Skymodel`:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "dataset = MapDataset.read(\"dataset.fits.gz\")\n",
+    "dataset.models.extend(models_pntgaus)\n",
+    "print(dataset.models)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "As above, we can sample the events from this `Dataset` and the `Observation` object previously defined:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "sampler = MapDatasetEventSampler(random_state=0)\n",
+    "events = sampler.run(dataset, observation)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let's inspect the events:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "events.peek()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "counts = Map.create(frame=\"galactic\", skydir=(0, 0.), binsz=0.02, npix=(150, 150))\n",
+    "counts.fill_events(events)\n",
+    "counts.smooth(0.04 * u.deg).plot(add_cbar=True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Again, we can fit the sampled events to compare simulation and observation:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "dataset = MapDataset.read(\"dataset.fits.gz\")\n",
+    "models_sim_pntgaus = Models.read(\"gauss-pwl.yaml\")\n",
+    "\n",
+    "counts = Map.from_geom(WCS_GEOM)\n",
+    "counts.fill_events(events)\n",
+    "dataset.counts = counts\n",
+    "dataset.models.extend(models_sim_pntgaus)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "fit = Fit([dataset])\n",
+    "result = fit.run(optimize_opts={\"print_level\": 1})\n",
+    "print(result)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "result.parameters.to_table()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The results are excellent as all the observed parameters are consistent withing 1 sigma with the simulated ones."
+   ]
+  },
```

Comment:
Perhaps remove this section on extended source simulation and add it as suggested exercises? The methodology doesn't really change...

Reply:
Right! I leave only the point-like case and the template case.